src/librustc/mir/interpret/error.rs

```diff
@@ -468,6 +468,7 @@ pub enum UnsupportedOpInfo<'tcx> {
     HeapAllocNonPowerOfTwoAlignment(u64),
     ReadFromReturnPointer,
     PathNotFound(Vec<String>),
+    TransmuteSizeDiff(Ty<'tcx>, Ty<'tcx>),
```

Comment:
We shouldn't add new "unclassified" error kinds. This one looks like UB to me?

But, how can it even be possible to get around the check rustc is already doing that the size must match...?

Reply:
This happens for array lengths, which are evaluated before typeck prevents const eval from running.... and now I'm thinking I can maybe just abort if typeck tables have errors. I'll try to remove this variant again, but I'd rather not make such a profound change to how we do early const eval in this PR

Reply:
I think we should never run Miri on untypechecked code. That would mean we'd lose tons and tons of useful invariants -- Miri relies on code being well-typed in a *lot* of places. This would be an endless source of ICEs.

Reply:
yea, this is an issue since 1.0. I'll try to address it again, I think the query system may be up to it nowadays.